### ISSUE TITLE
Releasing prepared statements

### ISSUE THREAD

**Issue by benjie** [brianc/node-postgres#1889]
Is there an official way to release prepared statements that use the `pgClient.query({name: "...", ...})` interface?

I have a solution, but I'm a bit worried it could break in a semver-minor update as the interfaces I'm using may not be official; thus I'm not keen to rely on it without approval. My solution [manually runs `DEALLOCATE ${escapeIdentifier(name)}`, and deletes `pgClient.connection.parsedStatements[name]`](https://github.com/graphile/graphile-engine/blob/38acf2333ce6be17e5518b7646fc4de82a67a824/packages/graphile-build-pg/src/pgPrepareAndRun.js#L42-L44).

(Our project's prepared statements are highly dynamic, so we need to keep an LRU-style cap on them to prevent exhausting resources on the database server.)

**Comment by abenhamdine** [comment]
> Is there an official way to release prepared statements that use the pgClient.query({name: "...", ...}) interface?

AFAIK there's no api available to do it.  
It would be useful to add it.
One would argue agains it with the rationale that this driver is low-level by design, but given that prepared statements are already tracked internally in connection.parsedStatements, it seems logical to be allowed to remove it

**Comment by brianc** [comment]
agreed - it was a somewhat bad design decision on my part long ago to "automagically" track named prepared statements...and having no way to kinda "reset" them is not ideal.

**Comment by benjie** [comment]
@brianc If you address this, please don't break the interface above in a minor/patch release :heart: Specifically allow me to continue running the following so that pgClient will forget that that statement was prepared:

```js
delete pgClient.connection.parsedStatements[key];
```
